client/src/pages/BrowsePage.js

```diff
@@ -18,22 +17,37 @@ const Container = styled.div`
 `;
 
 export const BrowsePage = () => {
-  const { data, loading, error, doFetch } = useAsync(() => getData(all));
+  const [method, setMethod] = useState(getData);
+  const { data, loading, error, doFetch } = useAsync(() => method);
+  const [tag, setTag] = useState("");
 
   useEffect(() => {
     doFetch();
-  }, []);
+  }, [method]);
```

Comment:
short answer:
```suggestion
  }, [method, doFetch]);
```
long answer, you have to rewrite useAsync so you can give it the method in doFetch() like doFetch(getData)

Reply:
thanks for checking :)